app.js

```diff
@@ -437,8 +437,8 @@ exports.getOG = function (options, callback) {
 			callback(new Error('Error from server'), null);
 		} else {
 			if (options.encoding === null) {
-				if (charset(response.headers)) {
-					body = iconv.decode(body, charset(response.headers));
+				if (charset(response.headers, body)) {
+					body = iconv.decode(body, charset(response.headers, body));

```

Comment:
May I add one little request here? the charset module accepts a third argument peekSize and only looks for charset encoding in the first `peekSize` bytes of the body.

The default is 512 which is too small for most sites that don't have the right header set in http response and only set the right charset in html meta tag, the tag is usually after the first 512 bytes.

I think increasing it to something like 10K would be fine as scraping is mostly IO bound and further more regex should short circuit anyway.